test/Microsoft.AspNet.Mvc.Core.Test/ActionResults/ViewResultTest.cs

```diff
@@ -179,6 +180,13 @@ public class ViewResultTest
                            .Returns(viewEngine.Object);
             serviceProvider.Setup(p => p.GetService(typeof(ILogger<ViewResult>)))
                            .Returns(new Mock<ILogger<ViewResult>>().Object);
+            serviceProvider.Setup(s => s.GetService(typeof(IOptions<MvcOptions>)))
+                .Returns(() => {
+                    var optionsAccessor = new Mock<IOptions<MvcOptions>>();
```

Comment:
This would create a new instance of `IOptions<MvcOptions>` each time you invoke `GetService(typeof(IOptions<MvcOptions>))`. I realize the test scenario doesn't rely on this behavior, but it would be nicer to have it behave closer to the real scenario and have it return a single value all the time. Could we change this to not use a lambda?